src/main/java/org/example/Main.java

```diff
@@ -1,7 +1,165 @@
+import java.util.Scanner;
+
 public class Main {
+
+    private static String[][] tasks;
+    private static Scanner scanner;
+    static final String[] OPTIONS = {"add", "remove", "list", "exit"};
+
     public static void main(String[] args) {
-        System.out.println("Hello, World!");
+        scanner = new Scanner(System.in);
+        tasks = readTasks();
+
+        while (true) {
+            displayUserOptions();
+            switch (scanner.nextLine().toLowerCase()) {
+                case "add" -> tasks = addTask();
+                case "remove" -> tasks = removeTask();
+                case "list" -> listTasks();
+                case "exit" -> exitTasks();
+                default -> System.err.println("Please select a correct option!");
+            }
+        }
+    }
+
+    public static void displayUserOptions() {
+        System.out.println(ConsoleColors.BLUE_BOLD_BRIGHT + "Please select an option from the following:" + ConsoleColors.RESET);
+        for (String option : OPTIONS) {
+            System.out.println(ConsoleColors.WHITE_BRIGHT + option + ConsoleColors.RESET);
+        }
+        System.out.println(ConsoleColors.BLUE_BRIGHT + "Type in the selected option:" + ConsoleColors.RESET);
+    }
+
+    public static String[][] readTasks() {
+        final var path = Paths.get("tasks.csv");
+        try {
+            final var lines = Files.readAllLines(path);
+
+            tasks = new String[lines.size()][];
+            for (int i = 0; i < lines.size(); i++) {
+                tasks[i] = lines.get(i).split(",");
+            }
+        } catch (IOException e) {
+            System.err.println("Failed to load tasks.csv");
+            e.printStackTrace(System.err);
+            System.exit(-1);
+        }
+        return tasks;
+    }
+
+    public static String[][] addTask() {
+        System.out.println(ConsoleColors.GREEN_BOLD_BRIGHT + "Please enter the description of the task including importance: " + ConsoleColors.RESET);
+        final String description = scanner.nextLine();
+
+        String deadline;
+        while (true) {
+            System.out.println(ConsoleColors.GREEN_BOLD_BRIGHT + "Please enter the deadline(YYYY-MM-DD): " + ConsoleColors.RESET);
+            deadline = scanner.nextLine();
+            if (deadline.matches("^\\d{4}-\\d{2}-\\d{2}$")) {
+                final String[] splitDate = deadline.split("-");
+                final var month = Integer.parseInt(splitDate[1]);
+                final var day = Integer.parseInt(splitDate[2]);
+                if ((month >= 1 && month <= 12) && (day >= 1 && day <= 31)) {
+                    break;
+                } else {
+                    System.err.println("Please enter a valid month(1-12) or a valid day(1-31)");
+                }
+            } else {
+                System.err.println("Invalid format! Please use YYYY-MM-DD format!");
+            }
+        }
+
+        Boolean done = null;
+        while (done == null) {
```

Comment:
CLEAN-CODE: validacni smycku bych udelal vsude podobne a jednodussi na pochopeni je while(true).

```suggestion
        while (true) {
```

Reply:
Opraveno.